src/NEventStore.Persistence.MongoDB/MongoPersistenceEngine.cs

```diff
@@ -525,12 +526,20 @@ private void UpdateStreamHeadAsync(string bucketId, string streamId, int streamR
                                 .Inc(MongoStreamHeadFields.SnapshotRevision, 0)
                                 .Inc(MongoStreamHeadFields.Unsnapshotted, eventsCount),
                             UpdateFlags.Upsert);
-                    }
-                    catch (MongoDuplicateKeyException ex)
-                    {
-                        Logger.Warn("Duplicate key exception {0} when upserting the stream head {1} {2}.", ex, bucketId, streamId);
-                    }
-                }), null);
+
+                    });
+                }
+                catch (OutOfMemoryException ex)
+                {

```

Comment:
Since we can tolerate errors in updating streams, a global try catch that ignore all exception is probably acceptable and fix all problems.